randomwallpaper@iflow.space/randomWallpaperMenu.js

```diff
@@ -85,6 +85,7 @@ var RandomWallpaperMenu = class {
 		});
 
 		this.openSettings.connect("activate", () => {
+			// FIXME: Unhandled promise rejection. To suppress this warning, add an error handler to your promise chain with .catch() or a try-catch block around your await expression.
```

Comment:
I don't understand this todo right now. Where exactly is this unhandled promise rejection?

Reply:
When the UI part called through `Gio.DBus.session.call()` fails it will spit out the unhandled promise rejection error message.

I tested quite a bit around with this but wasn't able to tame it at that time. There might have been multiple similar errors happening at the same time. I *think* I have it now caught up with a simple `.catch()` but I still don't know why that didn't worked out at that time:

``` ts
Gio.DBus.session.call(…).catch(error => {
    this._logger.error(error);
});
```

https://github.com/Lucki/RandomWallpaperGnome3/blob/7411f62e971e95344eed23105dc666764ebf1c2b/src/randomWallpaperMenu.ts#L122-L133